packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -817,6 +817,13 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
       return;
     }
 
+    // Refresh the token cache to ensure we have the latest token metadata
+    // This fixes a bug where the cache from construction time could be stale/empty
+    const { tokensChainsCache } = this.messenger.call(
+      'TokenListController:getState',
+    );
```

Comment:
same here , we should fetch this value from the state to ensure we get last updated data